### PR TITLE
chore(core): validate NotificationCategory

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- validate the notification category values

### CODE REVIEW

File: core/api/src/services/notifications/convert.ts [hunk 1]
```diff
@@ -65,18 +63,18 @@ export const notificationCategoryToGrpcNotificationCategory = (
...
-    case GaloyNotificationCategories.AdminNotification:
+    case NotificationCategory.AdminNotification:
       return GrpcNotificationCategory.ADMIN_NOTIFICATION
-    case GaloyNotificationCategories.Balance:
+    case NotificationCategory.Balance:
       return GrpcNotificationCategory.BALANCE
-    default:
-      throw new Error(`Not implemented: ${category}`)
+    case NotificationCategory.Marketing:
+      return GrpcNotificationCategory.MARKETING
```
Comment (dolcalmi) [inline core/api/src/services/notifications/convert.ts:77]: it is better to keep the default returning an error.. specially if new items are added to the categories

File: core/api/src/services/notifications/convert.ts [hunk 2]
```diff
@@ -44,19 +44,17 @@ export const grpcNotificationCategoryToNotificationCategory = (
...
-      return GaloyNotificationCategories.Balance
+      return NotificationCategory.Balance
     case GrpcNotificationCategory.ADMIN_NOTIFICATION:
-      return GaloyNotificationCategories.AdminNotification
+      return NotificationCategory.AdminNotification
     default:
-      return new InvalidPushNotificationSettingError(
-        `Invalid notification category: ${category}`,
-      )
+      return new InvalidPushNotificationSettingError("Invalid notification category")
```
Comment (dolcalmi) [inline core/api/src/services/notifications/convert.ts:57]: why did you remove `${category}` also is this still only related to push notifications?

File: core/api/src/domain/notifications/index.ts [hunk 3]
```diff
@@ -17,11 +17,25 @@ export const NotificationType = {
...
 ): NotificationCategory | ValidationError => {
   // TODO: add validation
-  if (!notificationCategory) {
+  if (
+    !notificationCategory ||
+    !Object.values(NotificationCategory).find(
+      (category) => category === notificationCategory,
+    )
+  ) {
     return new InvalidPushNotificationSettingError("Invalid notification category")
```
Comment (dolcalmi) [inline core/api/src/domain/notifications/index.ts:39]: Maybe this is no more `InvalidPushNotificationSettingError`, probably require an specific error

**Review by dolcalmi — CHANGES_REQUESTED** [review]
looks good, just minor comments/changes

File: core/api/src/services/notifications/convert.ts (hunk 2 above)
Comment (UncleSamtoshi) [reply core/api/src/services/notifications/convert.ts:57]: Mistake, I'll add it back in

File: core/api/src/services/notifications/convert.ts (hunk 1 above)
Comment (UncleSamtoshi) [reply core/api/src/services/notifications/convert.ts:77]: Will add a default

File: core/api/src/domain/notifications/index.ts (hunk 3 above)
Comment (UncleSamtoshi) [reply core/api/src/domain/notifications/index.ts:39]: Added a more specific error

**Review by bodymindarts — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]
